fix: “기존 채널로 이동” 설정이 새로고침 시 초기화되는 문제 수정

Eun0713 opened the pull request:
### ✨ 이슈 번호 

- #110 

### 📌 설명

- 사용자가 settings 페이지에서 “기존 채널로 이동” 설정을 켜도 페이지를 새로고침하면 해당 설정이 다시 꺼진 상태로 초기화되는 문제를 수정한 Pull Request입니다.


### 📃 작업 사항

- [x] `useUserProfile` 훅을 Settings 페이지에 추가하여 사용자 설정값을 서버에서 불러오도록 수정
- [x] 새로고침 시에도 `isAdDetect`, `isReturnChannel` 설정값이 유지되도록 변경

### 💡 참고 사항 


https://github.com/user-attachments/assets/c219687e-d91d-4010-9a8d-0661ea93ccab

### 💭 리뷰 사항 반영 

- [x] persist 적용으로 불필요해진 useUserProfile 호출 제거
  - zustand persist로 설정값을 localStorage에 저장하도록 개선됨에 따라, Settings 페이지에서 더 이상 useUserProfile를 호출할 필요가    없어 해당 로직 제거
  
### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

Eun0713 committed:
fix: 새로고침 시 설정값이 default로 초기화되는 문제 해결

devjaeyoon commented on src/pages/Settings.jsx:
```diff
@@ -6,9 +6,14 @@ import {
   SETTING_EXPLANATIONS,
 } from "@/constants/settingOptions";
 
+import useUserProfile from "../hooks/useUserProfile";
```
절대경로 사용하면 더 좋을 것 같아요!

d0-neii replied:
네! 제가 확인을 제대로 못 했네요😅 바로 반영했습니다

Eun0713 committed:
chore: import 경로 절대 경로로 변경

devjaeyoon commented on src/pages/Settings.jsx:
```diff
@@ -5,10 +5,14 @@ import {
   const settingTypes = Object.values(SETTING_TYPES);
 
+  const userId = localStorage.getItem("userId");
+  useUserProfile(userId);
```
저는 `useUserProfile`이 api를 호출하기 때문에 `useUserStore`의 세팅 정보들을 persist하고 localStorage에 settings 정보가 있다면 localStorage에 있는 값을 가져오고 변경만 api 호출하는 식으로 진행하면 사용자가 설정 페이지에 진입할 때마다 불필요한 api 호출을 줄일 수 있다 생각했어요. 이 방식에 대해 두 분은 어떻게 생각하시는지 궁금합니다. @Eun0713 @faber-h

faber-h replied:
재윤님 말씀처럼 zustand 스토어에서 persist 옵션을 활용하는 방법에 동의해요!
속도도 빨라지고 네트워크 요청도 줄일 수 있을 것 같습니다!

d0-neii replied:
두 분 말씀 주신 내용 공감합니다! 
persist를 적용해서 설정값을 localStorage에 저장하는 방식 반영했습니다~
좋은 리뷰 감사합니다😊

Eun0713 committed:
feat: 사용자 설정 상태에 zustand persist 적용

Eun0713 committed:
refactor: persist 적용으로 불필요해진 useUserProfile 호출 제거

- zustand persist로 설정값을 localStorage에 저장하도록 개선됨에 따라
- Settings 페이지에서 더 이상 useUserProfile를 호출할 필요가 없어 해당 로직 제거

faber-h commented on src/store/useUserStore.js:
```diff
@@ -1,11 +1,19 @@
+        set({ settings: { isAdDetect, isReturnChannel } }),
+    }),
+    {
+      name: "user-settings",
```
`persist`를 사용해서 새로고침 시 사용자 설정이 유지되도록 처리를 잘 해주셨네요!
한 가지 제안드리자면, persist의 name을 기존에 사용하고 있는 `useChannelStore`의 `channel-storage` 네이밍과 일관성 있게 맞춰서 `user-storage`로 바꾸는 건 어떨까요?
네이밍이 일관되면 localStorage 관리할 때 더 직관적일 것 같습니다! 😊

d0-neii replied:
좋은 제안 감사합니다!
처음엔 저장값이 전부 유저의 설정 정보여서 `user-settings`로 설정했는데,
말씀해주신 것처럼 `channel-storage`와 일관되게 맞추는 게 더 직관적일 것 같아 `user-storage`로 변경했습니다.
그렇지만 현재 훅 이름이 `useUserStore`인 반면 실제 저장 대상은 설정 값뿐이라 훅 이름도 `useUserSettingsStore`처럼 리팩토링해보는 걸 추후 고려해봐도 좋을 것 같습니다 :)🙌

faber-h replied:
좋은 의견 감사합니다!
프로젝트가 아직 규모가 작고, DB에서도 user 테이블과 userSetting 테이블을 별도로 분리하지 않고 있어서 그에 맞춰 store도 useUserStore로 구성했었지만 말씀하신대로 현재 구조상 settings만 다루고 있으니 `useUserSettingsStore` 으로 리팩토링해도 좋을 것 같아요!
그리고 `channel-storage`와의 네이밍 일관성을 고려해서 `user-storage`로 수정해주신 것도 정말 좋았습니다!
네이밍이 일관되서 유지보수에 더 도움이 될 것 같아요. 😀

d0-neii replied:
[스토어 네이밍 명확화 (useUserStore → useUserSettingsStore)](https://github.com/Radio-Premium/RadioPremium-FE/issues/152)
말씀 주신 부분 이슈 등록해두고, `useUserSettingsStorage`로 변경하면서 `persist name`이랑 `import` 경로까지 정리해서 반영했습니다😊

faber-h reviewed (approved):
LGTM 🔥

devjaeyoon reviewed (approved):
LGTM😄
